model/UsersData.ts

```diff
@@ -0,0 +1,151 @@
+import * as fsPromise from "fs";
+import { User } from "./User";
+import { File } from "./File";
+import { input } from "../utility/constants";
+
+export class UsersData {
+  public usersData: User[];
+  constructor() {
+    //Loading data from disk to in-memory
+    const data = fsPromise.readFileSync("./database/db.txt", "utf8");
+    let allUsers: any = data.split(",").slice(1);
+
+    allUsers = allUsers.map((element: any) => {
+      let bufferObj = Buffer.from(element, "base64");
+      return JSON.parse(bufferObj?.toString("utf8"));
+    });
+    this.usersData = allUsers;
+  }
+
+  public getUserData() {
+    return this.usersData;
+  }
+
+  public displayUserDetails(inputChoice:number,sortOrder:boolean) {
+    if(inputChoice == 1){
+      this.usersData = this.sortByName(this.usersData,sortOrder);
+    }else if(inputChoice == 2){
+      this.usersData = this.sortByRoleNumber(this.usersData,sortOrder);
+    }else if(inputChoice == 3){
+      this.usersData = this.sortByAge(this.usersData, sortOrder);
+    }else {
+      this.usersData = this.sortByAddress(this.usersData,sortOrder);
+    }
+    const headers = ["name", "age", "address", "rollNumber", "courses"];
+    // Find maximum width for each column
+    const columnWidths = headers.map((header, index) => {
+      const maxWidth = Math.max(
+        header.length,
+        ...this.usersData.map((user: any) => {
+          if (header === "courses") {
+            return user[header].join(", ").length;
+          } else {
+            return String((user as any)[header]).length;
+          }
+        })
+      );
+      return maxWidth + 2; // Add padding for better readability
+    });
+    // Display column headers
+    console.log(
+      columnWidths
+        .map((width, index) => headers[index].padEnd(width))
+        .join(" | ")
+    );
+    // Display separator line
+    console.log(
+      "-".repeat(columnWidths.reduce((sum, width) => sum + width + 3, 0))
+    );
```

Comment:
Move this logic to utility file. It has nothing to do with this class.

Reply:
Done